uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/OneAuthTestApp.java

```diff
@@ -0,0 +1,200 @@
+        }
+
+        Logger.i(TAG, "Handle AAD Login page prompt..");
+        // handle prompt in AAD login page
+        new AadPromptHandler(promptHandlerParameters).handlePrompt(username, password);
+    }
+
+    public void handleBackButton() {
+        UiAutomatorUtils.pressBack();
+    }
+
+    @Override
+    public void onAccountAdded() {
+        // nothing needed here
+    }
+
+    @Override
+    public void confirmAccount(@NonNull final String username) {
+        // Make sure we are seeing the output text view
+        final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtGeneralInfo");
+        try {
+            Assert.assertTrue(resultUIObject.getText().contains("Result: Success"));
+        } catch (UiObjectNotFoundException e) {
+            throw new AssertionError("Could not click on object with txt general info text");
+        }
+    }
+
+
+    public void assertSuccess() {
+        try {
+            final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtGeneralInfo");
+            resultUIObject.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+
+            final UiObject accountIdObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtAccountId");
+            accountIdObject.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+
+            Assert.assertTrue(resultUIObject.getText().contains("Result: Success"));
+            Assert.assertFalse(TextUtils.isEmpty(resultUIObject.getText()));
+            Assert.assertFalse(TextUtils.isEmpty(accountIdObject.getText()));
+        } catch (UiObjectNotFoundException e) {
+            throw new AssertionError("Could not click on object general text and account id");
+        }
+    }
+
+    /**
+     * Returns string list of all accounts available
+     */
+    public List<String> getAllAccounts() {
+        List<String> accountsList = new ArrayList<>();
+        UiAutomatorUtils.handleButtonClick("com.microsoft.oneauth.testapp:id/get_all_accounts_button");
+        final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/all_accounts_list");
+        try {
+            int size = resultUIObject.getChildCount();
+            for (int i = 0; i < size; i++) {
+                UiObject object = resultUIObject.getChild(new UiSelector().clickable(true).index(i));
+                accountsList.add(object.getText());
+            }
+        } catch (UiObjectNotFoundException e) {
+            throw new AssertionError("Could not click on object showing list of accounts");
+        }
+        return accountsList;
+    }
+
+    public String getTokeSecret() {
+        final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtSecret");
+        try {
+            return resultUIObject.getText();
+        } catch (UiObjectNotFoundException e) {
```

Comment:
throw new AssertionError(e);

Reply:
We are passing custom message on UI object not found exception in entire code. Is there any reason for passing exception or custom message in other places.

Reply:
Not in the entire code. It must be in some places.

Reply:
Do we have a pattern or guideline that needs to be followed for passing custom message or exception in the AssertionError?